g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductComponentAction.java

```diff
@@ -180,5 +180,84 @@ public APIResponseDTO checkTranslationResult(String productName, String componen
 						en);
 		}
 		return m;
-	}	
+	}
+
+	/**
+	 * Get the the matched version from the version list by comparing to the input version
+	 *
+	 * @param productName
+	 * @param version
+	 * @return a matched version, if there's no matched version then return input version
+	 */
+	private String getMatchedVersion(final String productName, final String version) throws L3APIException{
+		Map<String, String[]> productsAndVersions = productService.getProductsAndVersions();
+		String matchedVersion = "";
+		if(productsAndVersions != null && !productsAndVersions.isEmpty()) {
+			String[] versionList = productsAndVersions.get(productName);
+			if(versionList != null && versionList.length > 0) {
+				for(String s : versionList) {
+					String f = filterVersion(s, version);
+					if(!compare(f, version) && compare(s, matchedVersion)) {
+                        matchedVersion = s;
+					}
+				}
+			}
+		}
+		if(!StringUtils.isEmpty(matchedVersion)) {
+			return matchedVersion;
+		} else {
+			return version;
+		}
+	}
+
+	/**
+	 * Compare source version and target version
+	 *
+	 * @param source
+	 * @param target
+	 * @return true means source > target
+	 */
+	private boolean compare(final String source, final String target) {
+		boolean b = false;
+		String[] s = source.split("\\.");
+		String[] t = target.split("\\.");
+		if(s.length == t.length) {
+			for(int i = 0; i < s.length; i++) {
+				if(Integer.parseInt(s[i]) == Integer.parseInt(t[i])) {
+					continue;
+				} else if(Integer.parseInt(s[i]) > Integer.parseInt(t[i])) {
+					b = true;
+				}
+				break;
+			}
+		}
+		if(!StringUtils.isEmpty(source) && StringUtils.isEmpty(target)) {
```

Comment:
Move this if to the beginning of the method?

Reply:
fixed.